ige/src/plugin/audio/AudioEngine.cpp

```diff
@@ -32,8 +33,23 @@ AudioEngine::AudioEngine(const char* deviceName)
     AudioEngine::get_native_exception();
 }
 
+AudioEngine::AudioEngine(AudioEngine&& engine)
+{
+    *this = std::move(engine);
+}
+
+AudioEngine& AudioEngine::operator=(AudioEngine&& other)
+{
+    this->m_context = other.m_context;
+    this->m_device = other.m_device;
+    other.m_moved = true;
+    return *this;
+}
```

Comment:
the context that is potentially held by `this` might not be closed here
`this->m_moved` is also never reset if it was set to `true`